### PR TITLE
#6: Popup view for Cadastrapp

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds the possibility of querying the cadastapp parcelle layer for land info and viewing the same in the popup viewer

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

##Issue

**What is the current behavior?**
#6 

**What is the new behavior?**
- Hovering on a parcelle layer will display a popup with GFI info catered to Cadastapp
- When a drawing tool is selected the popup will close with the map trigger mechanism reset

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by offtherailz — CHANGES_REQUESTED** [review]
It is good. Here a couple of things to refine: 

- [ ] Due to some changes, please solve conflicts.
- [ ] Deploying the extension in prod it looks to not be able to deactivate gfi correctly. Please check. Here the behaviour
Peek 01-02-2021 18-02
- [ ] when the popup is in an area that do not contain data, it displays the last data on the current point (that should not contain data) and an error notification. The popup should not appear in that case, or at least show a "no data here advice" i think an error notification is too hard for this interaction. The original application didn't show any new popup. I should proceed in the same way.
- [ ] When the mouse goes out of the view (e.g. on cadastrapp window) , the popup appears anyway. It also reloads if you put the mouse upon the popup. This because you are not intercepting the MOUSE_OUT event should be intercepted a little earlier. (takeUntil is applied after getFeature). Moreover i think the flow of `showPopupEpic` should be stopped by mouse_out too. If it is a problem, you could try to put the stream of `showPopupEpic` directly inside `mouseMoveMapEventEpic` instead of triggering the action, but also separate epics is ok ). 
- [ ] Please intercept also cadastrapp CLOSE event, other than mouse out, to prevent popup to show after cadastrapp has been closed.
- [ ] Differently from original app, when draw tools are enabled, the popup is disabled. This is because MapStore, when draw tools are enabled, disables the other interactions by default (e. g. get feature info).  This is fine IMHO, because showing the popup while drawing may interfer drawing operations.The only thing I don't find too much clear is that to re-activate the popup you have to turn off selection tools. @tdipisa Do you think we can add a selection tool for "inspection" near the draw point/line/polygon?

**Comment by dsuren1** [comment]
> * [ ]  Differently from original app, when draw tools are enabled, the popup is disabled. This is because MapStore, when draw tools are enabled, disables the other interactions by default (e. g. get feature info).  This is fine IMHO, because showing the popup while drawing may interfer drawing operations.The only thing I don't find too much clear is that to re-activate the popup you have to turn off selection tools. @tdipisa Do you think we can add a selection tool for "inspection" near the draw point/line/polygon?

@offtherailz  Apart from this, I have fixed the rest. Kindly give me your feedback. Thanks.
@tdipisa Please do let me know if we have to place a "inspection" button or the current flow if fine? Thanks!

File: assets/index.json [hunk 1]
```diff
@@ -1,9 +1,14 @@
 {
   "plugins": [
     {
-      "name": "Cadastrapp",
+      "name": "Cadastrapp1",
```
Comment (offtherailz) [inline assets/index.json:4]:
```suggestion
      "name": "Cadastrapp",
```

Comment (dsuren1) [reply assets/index.json]: I was testing by importing a different version. Forgot to revert it back. I will change it now :)

**Comment by offtherailz** [comment]
Thanks @dsuren1, I (fixed cadastrapp1 into cadastrapp and) installed on demo the current version, so tobia can easely undestand. 
@tdipisa: 
Differently from original app, when draw tools are enabled (e.g. while searching), the popup is disabled. This is because MapStore, when draw tools are enabled, disables the other interactions by default (e. g. get feature info). 

This is fine IMHO, because showing the popup while drawing may interfer drawing operations.The only thing I don't find too much clear is that to re-activate the popup you need to turn off selection tools, that is not so much explicit.
What do you think about adding another button (icon mouse, active by default on startup) that explicitly tells to turn on the popup (in mutual exclusion with drowing tools). I think this should result more user friendly. Please try the UI and tell me your feelings about it.

File: assets/index.json (hunk 1 above)
Comment (dsuren1) [reply assets/index.json]: Thanks @offtherailz

**Review by offtherailz — APPROVED** [review]

**Pull request merged**